feature: Add Initial gitlint Config

ChristianTackeGSI opened the pull request:
gitlint looks like an interesting tool to check conformance of git commit messages to some rules.

This initial config allows all previous commits of this repository.

It has the conventional commits prepared but not yet enabled.

ChristianTackeGSI committed:
feature: Add Initial gitlint Config

gitlint looks like an interesting tool to check conformance
of git commit messages to some rules.

This initial config allows all previous commits of this
repository.

It has the conventional commits prepared but not yet
enabled.

dennisklein commented:
Wow, it runs the push checks, too. Could you add a filter to the test workflow that - let's say branches with prefix `pr/` - do not see the push event?

ChristianTackeGSI committed:
feature(CI): Do not run CI for push to "pr/*" branches

We currently try to host our PR related branches directly
in the main repository. This has pros and cons.

It currently meant, that the CI runs twice: Once for the
push to the branch, once for the pull-request.

Let's name branches for PRs as "pr/*".

So exclude those branches from push events.

ChristianTackeGSI commented:
@dennisklein 
> Wow, it runs the push checks, too. Could you add a filter to the test workflow that - let's say branches with prefix pr/ - do not see the push event?

hopefully done.
